cwms-data-api/src/main/java/cwms/cda/data/dao/StreamLocationDao.java

```diff
@@ -88,9 +90,14 @@ public List<StreamLocation> retrieveStreamLocations(String streamIdMask, String
      */
     public StreamLocation retrieveStreamLocation(String officeId, String streamId, String locationId, String stationUnit, String stageUnit, String areaUnit) {
         return connectionResult(dsl, conn -> {
-            RETRIEVE_STREAM_LOCATION retrieveStreamLocation = CWMS_STREAM_PACKAGE.call_RETRIEVE_STREAM_LOCATION(DSL.using(conn).configuration(),
-                    locationId, streamId, stationUnit, stageUnit, areaUnit, officeId);
-            return fromJooqStreamLocation(retrieveStreamLocation, locationId, streamId, officeId, stationUnit, stageUnit, areaUnit);
+            try {
+                setOffice(conn, officeId);
+                RETRIEVE_STREAM_LOCATION retrieveStreamLocation = CWMS_STREAM_PACKAGE.call_RETRIEVE_STREAM_LOCATION(DSL.using(conn).configuration(),
+                        locationId, streamId, stationUnit, stageUnit, areaUnit, officeId);
+                return fromJooqStreamLocation(retrieveStreamLocation, locationId, streamId, officeId, stationUnit, stageUnit, areaUnit);
+            } catch (DataAccessException e) {
+                throw new NotFoundException("Stream location " + locationId + " not found on stream " + streamId);
```

Comment:
include office id in the message

Reply:
Updated this to use wrapException

Reply:
removed wrapException. Not needed since error code was added.